packages/backend/src/core/balances/BalanceProvider.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  AssetId,
+  ChainId,
+  EthereumAddress,
+  UnixTime,
+} from '@l2beat/shared-pure'
+
+import { BalanceRecord } from '../../peripherals/database/BalanceRepository'
+import { BalanceCall } from '../../peripherals/ethereum/calls/BalanceCall'
+import { MulticallClient } from '../../peripherals/ethereum/MulticallClient'
+
+export interface HeldAsset {
+  holder: EthereumAddress
+  assetId: AssetId
+}
+
+export interface BalanceProvider {
+  fetchBalances: (
+    missingAssets: HeldAsset[],
+    timestamp: UnixTime,
+    blockNumber: number,
+  ) => Promise<BalanceRecord[]>
+  getChainId: () => ChainId
+}
+
+export class EthereumBalanceProvider implements BalanceProvider {
```

Comment:
This seems like it's only going to be using Ethereum mainnet, so taking `chainId` as argument seems useless since it will always be equal to `ChainId.ETHERUM`. If that is not the case then the naming of this class is misleading

Reply:
https://github.com/l2beat/l2beat/pull/1630/commits/400987444f2034b3e4d97f61b345f5a55698a1f7

Reply:
Fair enough, worth mentioning that such thing will shape itself along the way once we start to tackle new 'connectors' to L2